From 78173e815ef5630cf1159efa958884082d0c8e21 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sun, 25 Feb 2018 01:39:32 -0500
Subject: [PATCH 007/101] enable ARM64_SW_TTBR0_PAN by default

---
 arch/arm64/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 393d71124f5d2..d7d8847a21290 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -1715,6 +1715,7 @@ config ARM64_SW_TTBR0_PAN
 	bool "Emulate Privileged Access Never using TTBR0_EL1 switching"
 	depends on !KCSAN
 	select ARM64_PAN
+	default y
 	help
 	  Enabling this option prevents the kernel from accessing
 	  user-space memory directly by pointing TTBR0_EL1 to a reserved
-- 
2.50.1

