From 24f9a18b998bb62a85ae3ac734aee9fba40df913 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 4 May 2017 18:11:31 -0400
Subject: [PATCH 015/101] stop hiding X86_16BIT behind EXPERT

---
 arch/x86/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 05880301212e3..6d18b36dc3b58 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -1233,7 +1233,7 @@ config VM86
 	default X86_LEGACY_VM86
 
 config X86_16BIT
-	bool "Enable support for 16-bit segments" if EXPERT
+	bool "Enable support for 16-bit segments"
 	default y
 	depends on MODIFY_LDT_SYSCALL
 	help
-- 
2.51.2

