From e2d8d4a5dcf9955efc17031995c0b0ce635757ee Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Tue, 22 Dec 2020 23:41:32 +0100
Subject: [PATCH 043/101] disable UID16 by default

---
 init/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/init/Kconfig b/init/Kconfig
index afb65eab2647e..fa0059a100beb 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1599,7 +1599,6 @@ menuconfig EXPERT
 config UID16
 	bool "Enable 16-bit UID system calls"
 	depends on HAVE_UID16 && MULTIUSER
-	default y
 	help
 	  This enables the legacy 16-bit UID syscall wrappers.
 
-- 
2.51.0

