From fe993c5084902a0373ea3a0101525b03aa94e15b Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Tue, 12 Dec 2023 18:35:05 +0100
Subject: [PATCH 046/101] mm/slab: enable RANDOM_KMALLOC_CACHES by default

---
 mm/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/Kconfig b/mm/Kconfig
index ed74fb55ef79b..4a05c2eb57776 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -286,7 +286,7 @@ config SLUB_CPU_PARTIAL
 	  Typically one would choose no for a realtime system.
 
 config RANDOM_KMALLOC_CACHES
-	default n
+	default y
 	depends on !SLUB_TINY
 	bool "Randomize slab caches for normal kmalloc"
 	help
-- 
2.51.2

