From 2364c96bca8764a4cc89888d2797dcbe4ac972c1 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Tue, 12 Dec 2023 18:35:05 +0100
Subject: [PATCH 046/101] mm/slab: enable RANDOM_KMALLOC_CACHES by default

---
 mm/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/Kconfig b/mm/Kconfig
index 770d1ea89bfb3..ea6af5d1a559e 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -286,7 +286,7 @@ config SLUB_CPU_PARTIAL
 	  Typically one would choose no for a realtime system.
 
 config RANDOM_KMALLOC_CACHES
-	default n
+	default y
 	depends on !SLUB_TINY
 	bool "Randomize slab caches for normal kmalloc"
 	help
-- 
2.50.1

