From 0e48f2c52479ff7147c88d2f93a59a10728341a0 Mon Sep 17 00:00:00 2001
From: randomhydrosol <randomhydrosol@glassrom.org>
Date: Mon, 14 Apr 2025 22:21:48 +0530
Subject: [PATCH 101/101] add toggle to always enable FINEIBT-BHI

Signed-off-by: randomhydrosol <randomhydrosol@glassrom.org>
---
 arch/x86/Kconfig              | 15 +++++++++++++++
 arch/x86/kernel/alternative.c |  7 ++++++-
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 9796d0e8cb538..892f675c26524 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -2441,6 +2441,21 @@ config FINEIBT_PARANOID_ALWAYS_ON
       provides lower-latency, more secure transitions between privilege levels.
       When supported, FRED can further enhance system performance and robustness.
 
+config FINEIBT_BHI_ALWAYS_ON
+    bool "Force FineIBT-BHI Always-On"
+    depends on FINEIBT_BHI
+    default n
+    help
+      When enabled, this option activates the complementary FineIBT-BHI mitigation by
+      default on supported platforms.
+
+      While the kernel includes a native mitigation for branch history injection (BHI)
+      attacks, on some modern platforms that native approach may not be fully adequate.
+      FineIBT introduces an additional mitigation layer—FineIBT-BHI—that complements the
+      native protection. Enabling this option forces the enhanced, complementary mitigation
+      into effect by default, thereby providing improved security against BHI-based
+      exploitation techniques.
+
 config HAVE_CALL_THUNKS
 	def_bool y
 	depends on CC_HAS_ENTRY_PADDING && MITIGATION_RETHUNK && OBJTOOL
diff --git a/arch/x86/kernel/alternative.c b/arch/x86/kernel/alternative.c
index edb285342c713..cfb8857d466b6 100644
--- a/arch/x86/kernel/alternative.c
+++ b/arch/x86/kernel/alternative.c
@@ -1179,8 +1179,12 @@ void __init_or_module apply_seal_endbr(s32 *start, s32 *end) { }
 enum cfi_mode cfi_mode __ro_after_init = __CFI_DEFAULT;
 
 #ifdef CONFIG_FINEIBT_BHI
+#ifdef CONFIG_FINEIBT_BHI_ALWAYS_ON
+bool cfi_bhi __ro_after_init = true;
+#else
 bool cfi_bhi __ro_after_init = false;
 #endif
+#endif
 
 #ifdef CONFIG_CFI_CLANG
 u32 cfi_get_func_hash(void *func)
@@ -1274,7 +1278,8 @@ static __init int cfi_parse_cmdline(char *str)
 			} else {
 				pr_err("Ignoring paranoid; depends on fineibt.\n");
 			}
-		} else if (!strcmp(str, "bhi")) {
+		} else if (!strcmp(str, "bhi") ||
+				IS_ENABLED(CONFIG_FINEIBT_BHI_ALWAYS_ON)) {
 #ifdef CONFIG_FINEIBT_BHI
 			if (cfi_mode == CFI_FINEIBT) {
 				cfi_bhi = true;
-- 
2.51.2

